libnmstate/schemas/operational-state.yaml

```diff
@@ -287,6 +287,23 @@ definitions:
                     type: string
                   access-tag:
                     type: string
+                link-aggregation:
+                  type: object
+                  required:
+                  - mode
+                  - slaves
+                  properties:
+                    mode:
+                      type: string
+                      enum:
+                      - balance-slb
+                    slaves:
+                      type: array
+                      minItems: 2
+                      items:
+                        type: object
+                        properties:
```

Comment:
Can you please give an example of an additional property? If you can add one now, it will make this much clearer why this layer is needed (instead of a list of iface names).

Reply:
There is a **lot** of OVS interface attributes available http://www.openvswitch.org/support/dist-docs/ovs-vswitchd.conf.db.5.html. However, we may just set them in a separate interface entry. We can change it to list of strings to match the linux bonding (although it is scary to limit ourselves).

Reply:
You may misunderstood me.
I was looking for an argument to leave it like this. It will be hard to change the api from a list of strings to list of objects in the future.

Reply:
I don't have any additional attribute to add now.

Reply:
I actually remember what I thought at the time of that review.
I thought that we can define the ports as they now and can set link aggregation as a group of other ports. That will work with keeping the slaves as strings.
Makes sense?

Reply:
I am strongly against that (as I raised with you before). The main issue is that the `port` entry we have now has attributes of an OVS `port`, e.g. VLAN.

If I understood you correctly, you want to have `link-aggregation` type linking multiple of these `ports`. That would mean `port` alone is an OVS `port` and `interface`, while `port` under `link-aggregation` is an OVS `interface` only, although it has all attributes of a port. It would make the API way more complicated and hard to read what is defined in a desired state.

Reply:
I think you got it right. The suggestion is merge port anf interface settings.. which fits all cases except bonding. For bonding, the enslaved ports will ignore the port settings. I think this is a better model.

Reply:
You want to setup interface details? Do it in respective interface `ethernet` or `ovs-internal`. OVS `bridge` `port` list should have `port` specific attributes only. There is this shorthand of "if you defined only name, we implicitly add an interface with the same name" which does not fit the OVS DB schema.

What do you think is a better model again?

Reply:
I think ovs has interface properties that apply to eth0 and alike as well. Those are relevant for ovs, not for eth0.
I will try to fetch an example later today.

Reply:
Please also try to write an example of the API proposed by you.